### PR TITLE
Webdriver browsing contexts not pipelines

### PR DESCRIPTION
**Pull request by asajeffrey**
At the moment, a webdriver session stores a `pipeline_id`s, which causes a mismatch with the spec, which asks a session to store a browsing context and a top-level browsing context. This PR fixes this mismatch.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because we are not testing webdriver

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16997)

### CODE REVIEW

**Comment by asajeffrey** [comment]
The first commit is from #16916.

**Comment by asajeffrey** [comment]
This is part of the ongoing effort to introduce top-level browsing context ids, in order to support multiple windows (#13994). cc @jdm @cbrewster @paulrouget

**Comment by asajeffrey** [comment]
IRC conversation with @jgraham: http://logs.glob.uno/?c=mozilla%23servo&s=22+May+2017&e=22+May+2017#c678807 and http://logs.glob.uno/?c=mozilla%23servo&s=22+May+2017&e=22+May+2017#c678945.

TL;DR: argh testing! (I ended up driving servo by hand with curl, got it to navigate, traverse and execute a script.)

**Comment by asajeffrey** [comment]
#16916 has landed, so this PR is next in the queue.

**Comment by asajeffrey** [comment]
@jgraham are you OK reviewing this or should I find another victim?

**Comment by jgraham** [comment]
I'm not an expert on all of this code, but the changes seem quite reasonable.

---

Reviewed 12 of 12 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[components/constellation/constellation.rs, line 906 at r1](https://reviewable.io:443/reviews/servo/servo/16997#-Kl4DswOGVqO7ZAn3Rgd:-Kl4DswOGVqO7ZAn3Rge:bi2j249) ([raw file](https://github.com/servo/servo/blob/79743b5358c989da23c11c025509dd2c46dba48b/components/constellation/constellation.rs#L906)):*
> ```Rust
>             }
>             // Handle a forward or back request
>             // Handle a forward or back request
> ```

This seems to be duplicated.

---

*[components/script/dom/htmliframeelement.rs, line 501 at r1](https://reviewable.io:443/reviews/servo/servo/16997#-Kl4L3N-dfwtvVtQLzBe:-Kl4L3N1dLJy6z3OTQCf:bxd0fu9) ([raw file](https://github.com/servo/servo/blob/79743b5358c989da23c11c025509dd2c46dba48b/components/script/dom/htmliframeelement.rs#L501)):*
> ```Rust
>         MozBrowserEvent::LocationChange(url, can_go_back, can_go_forward) => {
>             BrowserElementLocationChangeEventDetail {
>                 url: Some(DOMString::from(url.as_str())),
> ```

Seems maybe unrelated?

---

*[components/webdriver_server/lib.rs, line 785 at r1](https://reviewable.io:443/reviews/servo/servo/16997#-Kl4Ps4E4eVufx04CJB9:-Kl4Ps4E4eVufx04CJBA:btx6nkl) ([raw file](https://github.com/servo/servo/blob/79743b5358c989da23c11c025509dd2c46dba48b/components/webdriver_server/lib.rs#L785)):*
> ```Rust
>             Err(WebDriverError::new(ErrorStatus::UnsupportedOperation, "Failed to convert keycodes"))));
> 
>         // TODO: there's a race condition caused by these being two separate messages,
> ```

It's not quite clear what "these" means in this comment.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/16997#-:-Kl4PytwVLQWcmUUkUli:b85w9pb)*

**Comment by asajeffrey** [comment]
Thanks! I made changes based on your comments.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[components/constellation/constellation.rs, line 906 at r1](https://reviewable.io:443/reviews/servo/servo/16997#-Kl4DswOGVqO7ZAn3Rgd:-Kl4oqozwqQmz4MNJebU:b-t8fyss) ([raw file](https://github.com/servo/servo/blob/79743b5358c989da23c11c025509dd2c46dba48b/components/constellation/constellation.rs#L906)):*
<details><summary><i>Previously, jgraham wrote…</i></summary><blockquote>

This seems to be duplicated.
</blockquote></details>

Oops. Fixed.

---

*[components/script/dom/htmliframeelement.rs, line 501 at r1](https://reviewable.io:443/reviews/servo/servo/16997#-Kl4L3N-dfwtvVtQLzBe:-Kl4pxvahukR51fAN_3b:bcn4mst) ([raw file](https://github.com/servo/servo/blob/79743b5358c989da23c11c025509dd2c46dba48b/components/script/dom/htmliframeelement.rs#L501)):*
<details><summary><i>Previously, jgraham wrote…</i></summary><blockquote>

Seems maybe unrelated?
</blockquote></details>

OK, I reverted this change.

---

*[components/webdriver_server/lib.rs, line 785 at r1](https://reviewable.io:443/reviews/servo/servo/16997#-Kl4Ps4E4eVufx04CJB9:-Kl4qPmKRovwmU_xB2t5:bt2vr1b) ([raw file](https://github.com/servo/servo/blob/79743b5358c989da23c11c025509dd2c46dba48b/components/webdriver_server/lib.rs#L785)):*
<details><summary><i>Previously, jgraham wrote…</i></summary><blockquote>

It's not quite clear what "these" means in this comment.
</blockquote></details>

Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/16997#-:-Kl4qdq-WSUF2xHgI4Id:b1edlp)*

**Comment by jgraham** [comment]
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/16997)*
